homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -150,21 +153,25 @@ class DeviceConnectivity(SurePetcareBinarySensor):
     def __init__(
         self,
         _id: int,
-        spc: SurePetcareAPI,
+        coordinator: DataUpdateCoordinator,
     ) -> None:
         """Initialize a Sure Petcare Device."""
-        super().__init__(_id, spc, DEVICE_CLASS_CONNECTIVITY)
+        super().__init__(_id, coordinator, DEVICE_CLASS_CONNECTIVITY)
         self._attr_name = f"{self.name}_connectivity"
         self._attr_unique_id = (
-            f"{self._spc.states[self._id].household_id}-{self._id}-connectivity"
+            f"{self.coordinator.data[self._id].household_id}-{self._id}-connectivity"
         )
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return super().available and bool(self._attr_is_on)
```

Comment:
```suggestion
```

Does the `available` provided by `CoordinatorEntity` need to be overwritten here?


Usually we only set `available` based on if we cannot communicate with the api or the service cannot communicate with the device. Since this sensor is connectivity, it seems like it should be available as long as the api is working otherwise it can never be off.

Reply:
Ok, make sense